sdk/rust-examples/huffman-encoding/src/main.rs

```diff
@@ -0,0 +1,165 @@
+//! Huffman Encoding Example
+//!
+//! ## Context
+//!
+//! Reads user input, and then encodes the file to binary format to compress it and save space.
+//! And then, it prints the encoded file.
+//! After that, the encoded file gets converted back to the original text, and that is output too.
+//! WARNING: Program is only desinged for ASCII characters, not UTF-8 encoding !!
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Copyright
+//!
+//! See the file `LICENSING.markdown` in the Veracruz root directory for licensing and
+//! copyright information.
+
+use std::fs ;
+use anyhow::{self, Ok} ;
+
+// For file input and output
+const INPUT_FILENAME: &'static str = "/input/huffman_example_input.txt" ;
+const OUTPUT_FILENAME: &'static str = "/output/encoded_output.txt" ;
+
+
+// For the Algorithm
+use std::collections::HashMap ;
+
+type Link = Option<Box<Node>> ;
+
+#[derive(Debug)]
+struct Node {
+    freq: i32,
+    ch: Option<char>,
+    
+    left: Link,
+    right: Link
+}
+    
+fn new_node(freq: i32, ch: Option<char>) -> Node {
+    Node {
+        freq: freq,
+        ch: ch,
+
+        left: None,
+        right: None,
+    }
+}
+
+fn new_box(n: Node) -> Box<Node> {
```

Comment:
This function can be deleted: just use `Box::new(-)` in its stead.

Reply:
fixed this one! I was really using an unnecessary function there😅